docs/architecture/deep-dives/state/index.md

```diff
@@ -47,75 +56,94 @@ export const MY_DOMAIN_DISK = new StateDefinition("myDomain", "disk");
 string literal `"disk"` or `"memory"` dictating where all the state using this `StateDefinition`
-should be stored. The Platform team will be responsible to reviewing all new and updated entries in
-this file and will be looking to make sure that there are no duplicate entries containing the same
-state name and state location. Teams _can_ have the same state name used for both `"disk"` and
-`"memory"` locations. Tests are included to ensure this uniqueness and core naming guidelines so you
-can ensure a review for a new `StateDefinition` entry can be done promptly and with very few
-surprises.
+should be stored.
 
-_TODO: Make tests_
+The Platform team will be responsible to reviewing all new and updated entries in this file and will
+be looking to make sure that there are no duplicate entries containing the same state name and state
+location. Teams _can_ have the same state name used for both `"disk"` and `"memory"` locations.
+Tests are included to ensure this uniqueness and core naming guidelines so you can ensure a review
+for a new `StateDefinition` entry can be done promptly and with very few surprises.
 
-:::note
+##### Client-specific storage locations
 
-Secure storage is not currently supported as a storage location in the State Provider Framework. For
-now, don't migrate data that is stored in secure storage but please contact the Platform team when you have
-data you wanted to migrate so we can prioritize a long-term solution. If you need new data in secure
-storage, use `StateService` for now.
+An optional third parameter to the `StateDefinition` constructor is provided if you need to specify
+client-specific storage location for your state.
 
-:::
+This will most commonly be used to handle the distinction between session and local storage on the
+web client. The default `"disk"` storage for the web client is session storage, and local storage
+can be specified by defining your state as:
+
+```typescript
+export const MY_DOMAIN_DISK = new StateDefinition("myDomain", "disk", { web: "disk-local" });
+```
 
-#### `KeyDefinition`
+#### `KeyDefinition` and `UserKeyDefinition`
 
-`KeyDefinition` builds on the idea of [`StateDefinition`](#statedefinition) but it gets more
-specific about the data to be stored. `KeyDefinition`s can also be instantiated in your own team's
-code. This might mean creating it in the same file as the service you plan to consume it or you may
-want to have a single `key-definitions.ts` file that contains all the entries for your team. Some
-example instantiations are:
+`KeyDefinition` and `UserKeyDefinition` build on the [`StateDefinition`](#statedefinition),
+specifying a single element of state data within the `StateDefinition`.
+
+The framework provides both `KeyDefinition` and `UserKeyDefinition` for teams to use. The
+`UserKeyDefinition` should be used for defining pieces of state that are scoped at a user level.
```

Comment:
⛏️ Consider stating this as an instruction. "Use `UserKeyDefinition` for state scoped to a user and `KeyDefinition` for user-independent state." or similar.